pkg/swiftring/funcs.go

```diff
@@ -57,18 +56,16 @@ func DeviceList(ctx context.Context, h *helper.Helper, instance *swiftv1beta1.Sw
 			cn := fmt.Sprintf("%s-%s-%d", swift.ClaimName, storageInstance.Name, replica)
 			foundClaim := &corev1.PersistentVolumeClaim{}
 			err = h.GetClient().Get(ctx, types.NamespacedName{Name: cn, Namespace: storageInstance.Namespace}, foundClaim)
-			capacity := resource.MustParse(storageInstance.Spec.StorageRequest)
-			weight, _ := capacity.AsInt64()
-			if err == nil {
-				if foundClaim.Status.Phase != corev1.ClaimBound {
-					err = fmt.Errorf("PVC %s found, but not bound yet (%s). Requeueing", cn, foundClaim.Status.Phase)
-					return "", "", err // requeueing
-				}
-				capacity := foundClaim.Status.Capacity[corev1.ResourceStorage]
-				weight, _ = capacity.AsInt64()
-			} else {
-				h.GetLogger().Info(fmt.Sprintf("Did not find PVC %s, assuming %s as capacity", cn, storageInstance.Spec.StorageRequest))
+			if err != nil {
+				return "", "", err // requeueing
 			}
+
+			if foundClaim.Status.Phase != corev1.ClaimBound {
+				err = fmt.Errorf("PersistentVolumeClaim %s found, but not bound yet (%s). Requeueing", cn, foundClaim.Status.Phase)
+				return "", "", err // requeueing
+			}
```

Comment:
Are these `err`s surfaced in a condition somewhere (i.e. in whatever reconcile function calls this)?  Just curious, as it would be good to have them appear in the CR status and/or CLI so that the user can understand what is going wrong without having to look at operator logs.

Reply:
Good idea, adding conditions

Reply:
The added commit now updates the conditions as well:

```
NAME         STATUS   MESSAGE
swift-ring            
swift-ring   Unknown   Setup started
swift-ring   False     SwiftRing error occured PersistentVolumeClaim "srv-swift-storage-0" not found
swift-ring   False     SwiftRing error occured PersistentVolumeClaim srv-swift-storage-0 found, but not bound yet (Pending). Requeueing
swift-ring   True      Setup complete

```